### ISSUE TITLE
 How to evaluate the results of the matben_bench_exp_band dataset? 

### ISSUE THREAD

**Issue by pincher-chen** [MasterAI-EAM/Darwin#3]
I'm puzzled about the training and evaluation merits of the matben_bench_exp_band dataset. In the train.py code, the function
```
def make_supervised_data_module(tokenizer: transformers.PreTrainedTokenizer, data_args) -> Dict:
    """Make dataset and collator for supervised fine-tuning."""
    train_dataset = SupervisedDataset(tokenizer=tokenizer, data_path=data_args.data_path)
    data_collator = DataCollatorForSupervisedDataset(tokenizer=tokenizer)
    return dict(train_dataset=train_dataset, eval_dataset=None, data_collator=data_collator)
```
implies that the eval_dataset is not used, right? How to calculate the loss ? It seems that in evaluate_matbench.py, all inference results and label values are imported for comparison? If I set up train, val, and test JSON files, how should I handle them?

**Comment by piupiupiuu** [comment]
> I'm puzzled about the training and evaluation merits of the matben_bench_exp_band dataset. In the train.py code, the function
> 
> ```
> def make_supervised_data_module(tokenizer: transformers.PreTrainedTokenizer, data_args) -> Dict:
>     """Make dataset and collator for supervised fine-tuning."""
>     train_dataset = SupervisedDataset(tokenizer=tokenizer, data_path=data_args.data_path)
>     data_collator = DataCollatorForSupervisedDataset(tokenizer=tokenizer)
>     return dict(train_dataset=train_dataset, eval_dataset=None, data_collator=data_collator)
> ```
> 
> implies that the eval_dataset is not used, right? How to calculate the loss ? It seems that in evaluate_matbench.py, all inference results and label values are imported for comparison? If I set up train, val, and test JSON files, how should I handle them?

Hi,
You can use train.py with train file to train the model first. 
After training, you can use evaluate_matbench.py to load the saved model and your val/test file to make predicition, evaluate_matbench.py will process the prediction output into the format ready to be evaluated.
Then you can apply evaluation metrics to evaluate the processed prediction output.

**Comment by 0xTong** [comment]
If you need any further assistance, pls let us know.
